guides/plugins/plugins/framework/caching/index.md

```diff
@@ -0,0 +1,250 @@
+---
+nav:
+  title: Caching
+  position: 20
+
+---
+
+# Caching
+
+Caching is a technique to store frequently accessed data in a temporary storage layer for faster retrieval, reducing latency and improving performance by avoiding repeated and costly data retrieval operations
+
+While caching enhances performance, it requires careful management of data consistency, cache invalidation strategies, and storage efficiency to prevent serving outdated or incorrect data.
+
+This guide will show you how you can modify the default caching mechanisms to suite your needs. If you are looking for information on how to add your routes to the HTTP-Cache take a look at [this guide](../../storefront/add-caching-to-custom-controller.md).
+
+## Cache Layers
+
+The current cache system of Shopware is based on a multi-layer system, in which the individual layers build on each other to improve performance and scalability.
+There is the [HTTP-Cache](../../../../../concepts/framework/http_cache.md) on the outer level and then multiple smaller internal "Object Caches" that are used to cache data in the application.
+
+For information on how to configure the different cache layers, please refer to the [caching hosting guide](../../../../hosting/performance/caches.md).
+
+### HTTP-Cache
+
+#### Modifying the cache keys
+
+Every cached item has a unique key that is used to retrieve the cached data later on. For that it is important that all the relevant information that have an effect on the data that is being cached is part of the key.
+For example if the same data is cached in multiple languages or for multiple sales channels the key must contain the language and sales channel information, so that the correct data can be retrieved later on.
+Please note that for every potential value your key part can take, a new cache entry will be created. So if you have a key part that can take 10 different values, you will have 10 times the amount of cache entries for the same data.
+
+If you add customization to your projects that lead to different versions of the page being rendered, you need to make sure that the cache key is unique for each version of the page. This can be done by adding a specific part to the cache key that shopware is generating.
+You can do so by subscribing to the `HttpCacheKeyEvent` event and add your specific part to the key.
```

Comment:
maybe add an
```
::: info
The event is called on any Request, make sure that you don't use expensive operations here like Database Queries and with an external reverse proxy, this needs to be done in the Configuration there also
:::
```

Reply:
To clarify and separate the two topics:

```
:::info
The event is called on any Request; make sure that you don't use expensive operations like Database Queries.

Also, with an external reverse proxy, the cache key might be generated on the proxy and not in your application. In that case, you need to add the key part to the reverse proxy configuration.
:::
```

wdyt ?

Reply:
done